demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -346,59 +391,188 @@ int main(int argc, char *argv[]) {
+                            for (const auto& request: userSpecifiedInferRequests) {
+                                emptyRequests.push(request);
+                            }
+                        } else emptyRequests.push(minLatencyInferRequest);
+
+                        modeInfo[prevMode].lastEndTime = std::chrono::high_resolution_clock::now();
+                        modeInfo[isUserSpecifiedMode] = ModeInfo();
+                    } else {
+                        presenter.handleKey(key);
+                    }
+                }
             }
+            else if (!emptyRequests.empty() && cap.isOpened()) {
+                auto startTime = std::chrono::high_resolution_clock::now();
+                
+                cv::Mat frame;
+                if (!cap.read(frame)) {
+                    if (frame.empty()) {
+                        if (FLAGS_loop_input) {
+                            cap.open((FLAGS_i == "cam") ? 0 : FLAGS_i.c_str());
+                        } else cap.release();
+                        continue;
+                    } else {
+                        throw std::logic_error("Failed to get frame from cv::VideoCapture");
+                    }
+                }
 
-            t1 = std::chrono::high_resolution_clock::now();
-            ocv_render_time = std::chrono::duration_cast<ms>(t1 - t0).count();
+                InferRequest::Ptr request;
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
 
-            if (isLastFrame) {
-                break;
-            }
+                    request = emptyRequests.front();
+                    emptyRequests.pop();
+                }
+                frameToBlob(frame, request, imageInputName);
+                
+                bool frameMode = isUserSpecifiedMode;
+                request->SetCompletionCallback([&mutex,
+                                                &completedRequestResults,
+                                                nextFrameId,
+                                                frame,
+                                                request,
+                                                outputName,
+                                                startTime,
+                                                frameMode,
+                                                &isUserSpecifiedMode,
+                                                &emptyRequests,
+                                                &callbackException,
+                                                &condVar] {
+                    {
+                        std::lock_guard<std::mutex> callbackLock(mutex);
+                    
+                        try {
+                            completedRequestResults.insert(
+                                std::pair<int, RequestResult>(nextFrameId, RequestResult{
+                                    frame, request->GetBlob(outputName)->buffer().as<float*>(),
```

Comment:
it should be rmap()

Reply:
There the problem with lifetime of `LockedMemory`. `outputMapped.as<float*>()` is stored in `RequestResult` while `outputMapped` is destroyed.